x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/optimizer/Optimizer.java

```diff
@@ -45,4 +59,85 @@ public LogicalPlan optimize(LogicalPlan verified) {
 
         return Arrays.asList(operators, label);
     }
+
+
+    private static class ReplaceWildcards extends OptimizerRule<Filter> {
+
+        private static boolean isWildcard(Expression expr) {
+            if (expr instanceof Literal) {
+                Literal l = (Literal) expr;
+                if (l.value() instanceof String) {
+                    String s = (String) l.value();
+                    return s.contains("*");
+                }
+            }
+
+            return false;
+        }
+
+        private static LikePattern toLikePattern(String s) {
+            // pick a character that is guaranteed not to be in the string, because it isn't allowed to escape itself
+            char escape = 1;
+
+            // replace wildcards with % and escape special characters
+            String likeString = s.replace("%", escape + "%")
+                .replace("_", escape + "_")
+                .replace("*", "%");
+
+            return new LikePattern(likeString, escape);
+        }
+
+        @Override
+        protected LogicalPlan rule(Filter filter) {
+            return filter.transformExpressionsUp(e -> {
+                // expr == "wildcard*phrase"
+                if (e instanceof Equals) {
```

Comment:
The blocks between Equals and NotEquals are the same and should be "compressed":
```java
if (e instanceof Equals || e instanceof NotEquals) {
 if (isWildcard ...) {
  Like like = new Like()...
  if (e instanceof NotEquals) {
       like = new Not (...);
   }
   e = like
}
return e;
```